plugins/spectate.cpp

```diff
@@ -31,25 +33,102 @@ using namespace df::enums;
 
 DFHACK_PLUGIN("spectate");
 DFHACK_PLUGIN_IS_ENABLED(enabled);
+bool dismiss_pause_events = false;
 bool following_dwarf = false;
+df::unit* our_dorf = nullptr;
 void* job_watched = nullptr;
 int32_t timestamp = -1;
 REQUIRE_GLOBAL(world);
 REQUIRE_GLOBAL(ui);
+REQUIRE_GLOBAL(pause_state);
+REQUIRE_GLOBAL(d_init);
+
+// todo: implement as user configurable variables
+#define tick_span 50
+#define base 0.99
+Pausing::AnnouncementLock* pause_lock = nullptr;
+bool lock_collision = false;
 
 command_result spectate (color_ostream &out, std::vector <std::string> & parameters);
 
 DFhackCExport command_result plugin_init (color_ostream &out, std::vector <PluginCommand> &commands) {
     commands.push_back(PluginCommand("spectate",
                                      "Automated spectator mode.",
-                                     spectate));
+                                     spectate,
+                                     false,
+                                     ""
+                                     " spectate\n"
+                                     "    displays plugin status\n"
+                                     " spectate enable\n"
+                                     " spectate disable\n"
+                                     " spectate auto-unpause\n"
+                                     "    toggle auto-dismissal of game pause events. e.g. a siege event pause\n"
+                                     "\n"));
+    pause_lock = World::AcquireAnnouncementPauseLock("spectate");
+    return CR_OK;
+}
+
+DFhackCExport command_result plugin_onstatechange(color_ostream &out, state_change_event event) {
+    if (enabled && world) {
+        switch (event) {
+            case SC_WORLD_LOADED:
+            case SC_MAP_LOADED:
+                if (dismiss_pause_events && World::ReadPauseState()) {
+                    *df::global::debug_nopause = false;
```

Comment:
There should also be a REQUIRE_GLOBAL() line added for this global

Reply:
Oh that actually can be removed now. I missed it on my spot check